BridgeAppSDK/SBAOnboardingManager.swift

```diff
@@ -55,15 +55,159 @@ public class SBAOnboardingManager: NSObject, SBASharedInfoController {
         self.init()
         self.sections = (dictionary["sections"] as? [AnyObject])?.map({ (obj) -> SBAOnboardingSection in
             return obj as! SBAOnboardingSection
-        })
+        }).sort({ sortOrder($0, $1) })
+    }
+    
+    /**
+     Returns an initialized task view controller for the given task type with this manager as its delegate.
+    */
+    public func initializeTaskViewController(onboardingTaskType: SBAOnboardingTaskType) -> SBATaskViewController? {
+        guard let sections = self.sections else { return nil }
+        
+        // Get the steps from the sections
+        let steps: [ORKStep] = sections.mapAndFilter({
+            stepsForSection($0, onboardingTaskType: onboardingTaskType)
+        }).flatMap({$0})
+        
+        // Create the task view controller
+        let task = SBANavigableOrderedTask(identifier: onboardingTaskType.rawValue, steps: steps)
+        let taskViewController = SBATaskViewController(task: task, taskRunUUID: nil)
+        taskViewController.delegate = self
+        
+        return taskViewController
     }
     
+    
+    // MARK: Protected methods that are exposed publicly to allow for override and testing
+    
+    /**
+     Convenience method for getting the section for a given section type.
+    */
     public func sectionForOnboardingSectionType(sectionType: SBAOnboardingSectionType) -> SBAOnboardingSection? {
         return self.sections?.findObject({ $0.onboardingSectionType == sectionType })
     }
     
-    public func factoryForSection(section: SBAOnboardingSection) -> SBASurveyFactory {
-        return section.defaultOnboardingSurveyFactory()
+    /**
+     Get the steps that should be included for a given `SBAOnboardingSection` and `SBAOnboardingTaskType`.
+     By default, this will return the steps created using the default onboarding survey factory for that section
+     or nil if the steps for that section should not be included for the given task.
+     @return    Optional array of `ORKStep`
+    */
+    public func stepsForSection(section: SBAOnboardingSection, onboardingTaskType: SBAOnboardingTaskType) -> [ORKStep]? {
+        
+        // Check to see that the steps for this section should be included
+        guard shouldIncludeSection(section, onboardingTaskType) else { return nil }
+        
+        // Get the default factory
+        let factory = section.defaultOnboardingSurveyFactory()
+        
+        // For consent, need to filter out steps that should not be included and group the steps into a substep. 
+        // This is to facilitate skipping reconsent for a user who is logining in where it is unknown whether
```

Comment:
logining maybe should be a word, but then you wouldn't need to follow it with 'in'